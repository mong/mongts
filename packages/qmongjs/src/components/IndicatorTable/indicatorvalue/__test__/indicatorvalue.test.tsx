import { render } from "@testing-library/react";

import { IndicatorValue } from "../";
import national_data from "../../../../test/test_data/national_data";

it("renders", () => {
  const data = national_data[1];

  const { container } = render(
    <table>
      <tbody>
        <tr>
          <IndicatorValue indicatorData={data} />
        </tr>
      </tbody>
    </table>
  );
  expect(container).toMatchSnapshot();
});

it("renders beregnet_andel", () => {
  const data = { ...national_data[1], type: "beregnet_andel" };
  const { container } = render(
    <table>
      <tbody>
        <tr>
          <IndicatorValue indicatorData={data} />
        </tr>
      </tbody>
    </table>
  );
  expect(container).toMatchSnapshot();
});
