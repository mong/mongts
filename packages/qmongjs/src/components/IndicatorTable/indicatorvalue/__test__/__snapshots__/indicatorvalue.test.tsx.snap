// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="selected_unit "
        >
          <div
            aria-label="Achieved level L"
            class="level"
          >
            <h4>
              11 %
              <i
                style="padding-left: 0.2em;"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: rgb(227, 7, 19); font-size: 1.2rem;"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
                  />
                </svg>
              </i>
            </h4>
          </div>
          <div
            class="summary"
          >
            300 av 2775
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders beregnet_andel 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td
          class="selected_unit "
        >
          <div
            aria-label="Achieved level L"
            class="level"
          >
            <h4>
              11 %
              <i
                style="padding-left: 0.2em;"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: rgb(227, 7, 19); font-size: 1.2rem;"
                  viewBox="0 0 512 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm0 448c-110.5 0-200-89.5-200-200S145.5 56 256 56s200 89.5 200 200-89.5 200-200 200z"
                  />
                </svg>
              </i>
            </h4>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
